optimum/intel/openvino/quantization.py

```diff
@@ -137,11 +140,83 @@ def quantize(
         >>> optimized_model = OVModelForSequenceClassification.from_pretrained("./quantized_model")
         ```
         """
+        if isinstance(self.model, OVBaseModel):
+            self._quantize_ovbasemodel(
+                calibration_dataset,
+                save_directory,
+                quantization_config,
+                file_name,
+                batch_size,
+                data_collator,
+                remove_unused_columns,
+                **kwargs,
+            )
+        elif isinstance(self.model, torch.nn.Module):
+            self._quantize_torchmodel(
+                calibration_dataset,
+                save_directory,
+                quantization_config,
+                file_name,
+                batch_size,
+                data_collator,
+                remove_unused_columns,
+                **kwargs,
+            )
+        else:
+            raise TypeError(f"Unsupported model type: {type(self.model)}")
+
+    def _quantize_ovbasemodel(
+        self,
+        calibration_dataset: Dataset,
+        save_directory: Union[str, Path],
+        quantization_config: OVConfig = None,
+        file_name: Optional[str] = None,
+        batch_size: int = 8,
+        data_collator: Optional[DataCollator] = None,
+        remove_unused_columns: bool = True,
+        **kwargs,
+    ):
+        save_directory = Path(save_directory)
+        save_directory.mkdir(parents=True, exist_ok=True)
+
+        calibration_dataloader = self._get_calibration_dataloader(
+            calibration_dataset=calibration_dataset,
+            batch_size=batch_size,
+            remove_unused_columns=remove_unused_columns,
+            data_collator=data_collator,
+        )
+
+        quantization_dataset = nncf.Dataset(calibration_dataloader, lambda x: x)
+        quantized_model = nncf.quantize(
+            self.model.model,
+            quantization_dataset,
```

Comment:
Is there a possibility to enable the pre-computed pkv depending on `config.use_cache` for text generation ?

Reply:
@echarlaix, I've added support of OVModelForCausalLM with pkv. I also need to change the way how the model export works (added `mo.convert_model()` for ONNX -> OV IR). This is a more safe way and we faced a lot of problems before when using read_model or ONNX directly.